### PR TITLE
Miscellaneous pre-release updates

### PR DESCRIPTION
**Pull request by zalimeni**
## Changes proposed in this PR:
- Add automated security scans a la other Consul repos
- Update CI Consul version and test workflow name
- Add CI fan-in jobs for required checks (updated in settings)
- Cancel concurrent CI workflows
- Add local testing guide to `CONTRIBUTING`
- Update changelog for Dependabot `x/net` PR
- Update Go to 1.21.9

## Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added

    [HashiCorp engineers only. Community PRs should not add a changelog entry.]::
    [Changelog entries should use present tense, e.g. "Add support for..."]::

### CODE REVIEW

File: scan.hcl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# Scan results are triaged via the GitHub Security tab for this repo.
+# See `security-scanner` docs for more information on how to add `triage` config
+# for specific results or to exclude paths.
+
+# .release/security-scan.hcl controls scanner config for release artifacts, which
+# unlike the scans configured here, will block releases in CRT.
+
+repository {
+  go_modules   = true
+  npm          = true
```
Comment (DanStough) [inline scan.hcl:19]: do we use `node` in this repo?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -22,6 +22,22 @@ If you're reporting what you think is a bug (i.e., something isn't right with an
...
+export CONSUL_LICENSE_PATH=...
+go test -v ./... -- -enterprise
+```
+
+To run the CE tests, omit the `-enterprise` flag and ensure your local binary is Consul CE.
+```shell
+go test -v ./...
+```
+
+Note that the CE tests cannot be run with a Consul Enterprise binary due to defaulting of tenancy fields.
```
Comment (DanStough) [inline CONTRIBUTING.md:40]: 💡 Good note!

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -238,3 +226,34 @@ jobs:
 
           echo "==> Push docker image $TAG"
           docker push "$TAG"
+
+  # This is job is required for branch protection as a required gihub check
```
Comment (DanStough) [inline .github/workflows/build.yml:230]:
```suggestion
  # This is job is required for branch protection as a required GitHub check
```

**Review by DanStough — APPROVED** [review]
👍 Thank you!

**Review by dduzgun-security — APPROVED** [review]

File: scan.hcl (hunk 1 above)
Comment (zalimeni) [reply scan.hcl:19]: Oops, copy 🍝 - thanks, removing.

File: CONTRIBUTING.md (hunk 2 above)
Comment (zalimeni) [reply CONTRIBUTING.md:40]: Learned the hard way 😅

**Pull request merged**